v-next/hardhat-node-test-runner/src/protocol.ts

```diff
@@ -0,0 +1,148 @@
+import type { HardhatRuntimeEnvironment } from "hardhat/types/hre";
```

Comment:
This, and the associated handlers may be so common that we should place them in core. I guess mocha running in parallel would also need them.